### PR TITLE
Fix JSON serialization of quantities with decimal values

### PR DESCRIPTION
**Pull request by rohahn**
Add IDecimalQuantity interface to expose the decimal value
Serialize decimal values as string to keep number of decimal places

### CODE REVIEW

**Comment by angularsen** [comment]
Solid work on this 👏 Love the tests!

I think it is clever to use string to support `decimal`, but I'm concerned we are breaking backwards compatibility on the JSON schema for those expecting to read `Value` as a number.

How about this:
- Keep `Unit` and `Value` exactly as before, the `double` representation of the number matching our common public interface for quantities.
- If value is not `double`, then add two more fields
  - `ValueString` with textual representation of number
  - `ValueType` with string values like `"decimal"`, `"float"`, `"SomeLibrary.BigInt"`

This makes the schema more extensible to other types of numbers with generics, as have been discussed a number of times:
#180
#285
#666
#714

**Comment by rohahn** [comment]
I implemented your proposal.

File: UnitsNet.Serialization.JsonNet/UnitsNetBaseJsonConverter.cs [hunk 1]
```diff
@@ -53,17 +73,28 @@ protected ValueUnit ReadValueUnit(JToken jsonToken)
...
-            if (valueUnit == null || string.IsNullOrWhiteSpace(valueUnit.Unit))
+            if (string.IsNullOrWhiteSpace(valueUnit?.Unit))
             {
                 return null;
             }
 
-            var unitParts = valueUnit.Unit.Split('.');
+            var unit = GetUnit(valueUnit.Unit);
+
+            return valueUnit switch
```
Comment (angularsen) [inline UnitsNet.Serialization.JsonNet/UnitsNetBaseJsonConverter.cs:83]: Nice

**Review by angularsen — APPROVED** [review]
Looks perfect to me, great work on this!

**Comment by angularsen** [comment]
Nuget is on the way out, I'd appreciate it if you gave it a spin to ensure it works as expected and comment back here! 🙇 
[Release JsonNet/4.3.0 · angularsen/UnitsNet](https://github.com/angularsen/UnitsNet/releases/tag/JsonNet%2F4.3.0)

**Comment by rohahn** [comment]
Why is the version number 4.3.0?

**Comment by angularsen** [comment]
JsonNet nuget is versioned separately from UnitsNet nuget, but they share major version.

**Pull request merged**